go-runtime/ftl/reflection/type_registry.go

```diff
@@ -49,13 +33,13 @@ func WithSumType[Discriminator any](variants ...Discriminator) TypeRegistryOptio
 			ref := TypeRefFromValue(v)
 			variantMap[ref.Name] = reflect.TypeOf(v)
 		}
-		t.RegisterSumType(reflect.TypeFor[Discriminator](), variantMap)
+		t.registerSumType(reflect.TypeFor[Discriminator](), variantMap)
 	}
 }
 
-// NewTypeRegistry creates a new [TypeRegistry] for instantiating types by their qualified
+// newTypeRegistry creates a new [TypeRegistry] for instantiating types by their qualified
 // name at runtime.
-func NewTypeRegistry(options ...TypeRegistryOption) *TypeRegistry {
+func newTypeRegistry(options ...TypeRegistryOption) *TypeRegistry {
```

Comment:
Ah nice, I wasn't thinking about making this private but it makes sense - unless we have a use case for type registry instances, we should make this private 👍